src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationAttributeStore.cs

```diff
@@ -230,8 +232,61 @@ internal bool TryGetPropertyStoreItem(string propertyName, [NotNullWhen(true)] o
                     propertyStoreItems[property.Name] = item;
                 }
 
+                return CombineMetadataTypeAttributes(propertyStoreItems);
+            }
+
+            [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
+            private Dictionary<string, PropertyStoreItem> CombineMetadataTypeAttributes(Dictionary<string, PropertyStoreItem> propertyStoreItems)
+            {
+                // check if metadata attribute is present on the type
+                MetadataTypeAttribute? attribute = ((MetadataTypeAttribute[])Attribute.GetCustomAttributes(_type, typeof(MetadataTypeAttribute))).FirstOrDefault();
+                if (attribute != null)
+                {
+                    // get properties of metadata type
+                    TypeInfo t = (TypeInfo)attribute.MetadataClassType;
```

Comment:
```suggestion
                    Type t = attribute.MetadataClassType;
```

There's no reason to use `TypeInfo` anymore. This class was an attempt at a redesign of Reflection during Silverlight and early .NET Core days. But with .NET Core 2.0 all the APIs on Type are back. Just use `Type`.

Reply:
Good to know. Corrected. Thanks.